buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -100,7 +100,21 @@ public void run() {
          *
          * See https://github.com/netty/netty/issues/3888.
          */
-        final int defaultMinNumArena = NettyRuntime.availableProcessors() * 2;
+        int defaultMinNumArena = NettyRuntime.availableProcessors() * 2;
+
+        // Respect MALLOC_ARENA_MAX if set.
+        final String mallocArenaMaxStr = System.getenv("MALLOC_ARENA_MAX");
+        if (mallocArenaMaxStr != null) {
+            try {
+                int value = Integer.parseInt(mallocArenaMaxStr);
+                if (value > 0) {
+                    defaultMinNumArena = Math.min(Integer.parseInt(mallocArenaMaxStr), defaultMinNumArena);
+                }
+            } catch (NumberFormatException ignore) {
+                // ignore
+            }
+        }
```

Comment:
Nit/style: this can be extracted to a separate method `getMallocArenaMax`, return `0` for `null`/error etc.